src/optimalcontrol/CMakeLists.txt

```diff
@@ -69,6 +72,22 @@ if (IDYNTREE_USES_IPOPT)
     list(APPEND INCLUDE_LIST ${IPOPT_INCLUDE_DIRS})
 endif()
 
+find_package(OsqpEigen QUIET)
```

Comment:
Can you add a `idyntree_handle_dependency(OsqpEigen)` line in https://github.com/robotology/idyntree/blob/devel/cmake/iDynTreeDependencies.cmake#L56 and use the `IDYNTREE_USES_OSQPEIGEN` to select if the dependency is used or not? Thanks!